src/Process/PlaceOnGroundProcessing.js

```diff
@@ -0,0 +1,126 @@
+import * as THREE from 'three';
+import Coordinates from '../Core/Geographic/Coordinates';
+import Extent from '../Core/Geographic/Extent';
+
+function tileAt(pt, tile) {
+    if (tile.extent) {
+        if (!tile.extent.pointIsInside(pt)) {
+            return undefined;
+        }
+
+        for (const c of tile.children) {
+            const t = tileAt(pt, c);
+            if (t) {
+                return t;
+            }
+        }
+        if (tile.materials[0].textures[0][0].coords.zoom > -1) {
+            return tile;
+        }
+        return undefined;
+    }
+}
+
+let _canvas;
+function _readTextureValueAt(layer, texture, x, y) {
+    if (texture.image.data) {
+        return texture.image.data[y * texture.image.width + x];
+    } else {
```

Comment:
In both case I'm reading values from a texture.
But this texture can be a data texture or an image based texture.